### ISSUE TITLE
Connection silently fails after removing certificate

### ISSUE THREAD

**Issue by mimi89999** [ChatSecure/ChatSecure-iOS#825]
After I remove a certificate that my server is using from pined certificates, connection will silently fail: when I select the connect button nothing happens. (I saw the connection in Prosody logs aborted just after the Hello.) The user should see a message that CS doesn't connect because the certificate was removed. Without that, I don't know what is happening.

**Comment by chrisballinger** [comment]
Yeah sorry this is a problem for a lot of people, especially LetsEncrypt

**Comment by chrisballinger** [comment]
Fix will be included in upcoming 4.1.1 release

**Comment by mimi89999** [comment]
@chrisballinger In latest TestFlight 4.1.1 (98) when I go to edit account and save/confirm, I will get a notification about accepting the new certificate, but  if I just click on the connect button, connection will still silently fail.

**Comment by chrisballinger** [comment]
Do you mean silently fails on "Log In" button on the initial account detail screen?

**Comment by mimi89999** [comment]
@chrisballinger This one
![img_1622](https://user-images.githubusercontent.com/8530546/28891743-d67c8712-77cb-11e7-8213-7e1a7c7b3798.PNG)

**Comment by mimi89999** [comment]
@chrisballinger This issue should be reopened.

**Comment by chrisballinger** [comment]
Ah I see, agreed. Should be a simple fix after my earlier refactor.

**Comment by mimi89999** [comment]
@chrisballinger Any news on this one? When certbot renews my certificate (by generating a new one...), ChatSecure silently break on my contacts devices. They are still receiving push notifications, but when they open the notification, they don't see the message, they also can't send messages.... I need to 'fix' it for them by clicking in the right place that will open that dialog for accepting the certificate....



I also think it is not a good idea to let users decide about the cert.
When somebody (like me) knows what is it and how to verify the certificate, it is a great security improvement. When somebody doesn't know, they have no clue what to do when they see the dialog...
I think it should be like in Conversations: by default CA valid certs are automatically accepted, but there is an expert setting that lets users decide for each cert.

**Comment by chrisballinger** [comment]
You're probably right about auto-trusting valid CAs by default, especially
now that Let's Encrypt is more widespread. There was a slight improvement
to the cert warning UX in a previous commit but I'm not sure if I released
it.

**Comment by mimi89999** [comment]
You didn't.

**Comment by mimi89999** [comment]
@chrisballinger Fixed in latest TestFlight [ChatSecure 4.1.1 (100)]. Thanks.

**Comment by mimi89999** [comment]
The current situation isn't good as after a certificate change, users continue to get push notifications, but after opening the app, they don't see any new messages and don't know they need to go to account settings and click on connect (to see the new cert dialog).

**Comment by chrisballinger** [comment]
Yeah, this could be improved. I think we will just go back to auto-trusting valid CA issued certs with a paranoid opt-out.

**Comment by mimi89999** [comment]
> I think we will just go back to auto-trusting valid CA issued certs with a paranoid opt-out.

I think it is the best solution.
